src/backends/plugins_compat/audio_plugin_compat.c

```diff
@@ -31,14 +31,14 @@
 #include "main/rom.h"
 #include "plugin/plugin.h"
 
-static void audio_plugin_set_frequency(void* aout, unsigned int frequency)
-{
-    #ifdef ENC_SUPPORT
-    if (Encoder_IsActive())
-        encoder_set_sample_rate(frequency);
-    #endif
-    
-    struct ai_controller* ai = (struct ai_controller*)aout;
+static void audio_plugin_set_frequency(void* aout, unsigned int frequency) {
+#ifdef ENC_SUPPORT
+    if (Encoder_IsActive() && rate_changed_callback != NULL)
```

Comment:
Encoder_IsActive currently depends on the backend encoder, we should change it to be dependent on a single bool. Also, we should probably respect M64+ naming conventions and, for example, change `rate_changed_callback` to `g_rate_changed_callback` since it's a global